compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -53,26 +54,29 @@ static llvm::cl::opt<bool> clDisableDistribution(
     llvm::cl::desc("disable thread distribution in codegen"),
     llvm::cl::init(false));
 
-static llvm::cl::list<int> clMmt4dDistributionTileSizes(
-    "iree-codegen-llvm-mmt4d-distribution-tile-sizes",
-    llvm::cl::desc("linalg.mmt4d distribution tile size"),
-    llvm::cl::ZeroOrMore);
-
-static llvm::cl::list<int>
-    mmt4dL1TileSizes("iree-codegen-llvm-mmt4d-l1-tile-size",
-                     llvm::cl::desc("linalg.mmt4d L1 tile size"),
-                     llvm::cl::ZeroOrMore);
-
-static llvm::cl::list<int>
-    mmt4dVectorSizes("iree-codegen-llvm-mmt4d-vector-size",
-                     llvm::cl::desc("linalg.mmt4d vector tile size"),
-                     llvm::cl::ZeroOrMore);
-
 static llvm::cl::opt<int>
     defaultDistTileSize("iree-codegen-llvm-distribution-size",
                         llvm::cl::desc("default distribution tile size"),
                         llvm::cl::init(64));
 
+static llvm::cl::opt<int> clNarrowMatmulTileBytes(
```

Comment:
Why two? just looking at the uses, one seems to be sufficient. Also would be good to make clear this is a developer and not meant to be used in any deployment.

Reply:
The PR description and the command line option description string explain the need for two separate controls. They are coincidentally the same now and might remain the same, but having them conflated will likely result in people having to turn a single knob that has a combination of wanted and unwanted effects.

Reply:
There already is a comment above these options saying it's experimental:
https://github.com/openxla/iree/pull/15651/files#diff-1ac727f2f0f42c0b7710d5c4778ec0b83e5a318765d2eb6d5960602a03f76f5dL40-L43

Maybe these options should have `-experimental` in their name, but that's a preexisting condition.

Reply:
> The PR description and the command line option description string explain the need for two separate controls. They are coincidentally the same now and might remain the same, but having them conflated will likely result in people having to turn a single knob that has a combination of wanted and unwanted effects.

Yeah, this is what I am saying though. This is a purely developer related flag /for debugging etc (I am also thinking of a single dispatch performance debugging). If we are trying to balance wanted/unwanted effects across model, using such flags is not going to be that much of a help in the long run.

Reply:
Actually there might be a misunderstanding here. I wouldn't call these flags "developer flags". Using them is currently necessary to tune IREE generated modules to (1) target details such as CPU cache sizes, and (2) runtime details such as number of threads used (which will affect choices of tile sizes, past a certain performance level).

It's very unfortunate that we currently require passing compile-time flags to tune for these runtime details, but it's the best that we can currently do as distribution tile sizes are compile-time constants (`lowering_config` attributes).

Given that constraint, it's the best *user* interface that we can do, except perhaps for inferring these values from higher-level command-line options for number of threads (that one actually exists just above in this file) and cache sizes, but as ultimately we shouldn't have such options in the compiler and we should be able to treat all distribution tile sizes as dynamic, I don't actually think that we should do that, and so I think this PR as-is is the best user interface that we can do.

Reply:
Consider that the existing options in this file, such as the `--iree-codegen-llvm-distribution-size` that was being used for mmt4d tile sizes prior to this PR, share these downsides without having the advantage that at least one can reason about what would be the right value to pass given some target/workload specifics.

Reply:
Re "single dispatch performance debugging", it really is the other way around:
* The existing `--iree-codegen-llvm-distribution-size` was geared towards single-dispatch tuning, as different dispatches would benefit from different tile sizes along the M/N dimension when they differ in their K dimension size and/or element types.
* The new flags introduced in this PR allow specifying a target tile size in bytes, that is the thing that can be meaningfully specified once for all dispatches, letting that resolve to different M/N dimension tile sizes for different dispatches.

Reply:
The existing options arent really meant to be anything apart from debugging flags, and they dont work as well :D . If you are doing some targeted debugging, there is a `compilation_info` that allows you to set things per mmt4d operation. As long this is for development, its OK, but this seems like trying to use this as part of a solution. If you really want two separate flags and that helps in debugging/path finding thats maybe OK, but thats all they are. They are for path finding.

Reply:
> Given that constraint, it's the best _user_ interface that we can do, except perhaps for inferring these values from higher-level command-line options for number of threads (that one actually exists just above in this file) and cache sizes, but as ultimately we shouldn't have such options in the compiler and we should be able to treat all distribution tile sizes as dynamic, I don't actually think that we should do that, and so I think this PR as-is is the best user interface that we can do.

This is kind of what I am afraid for. This is not a "user interface" at all...

Reply:
Even if they're just for path-finding, the two separate flags still help.  I would have an easier time understanding how that's for path-finding, though, if I understood what the post-path-finding replacement will be. Setting `compilation_info` manually in the IR isn't a user-centric solution.

Reply:
I mean, the basic hardware and workload requirements tell us what the post-path-finding solution will be: all distribution tile sizes need to be dynamic. So `compilation_info` can only be another iteration of path-finding. I don't know the prioritization and the path toward that, though, so for the time being I treat this as the best we can offer any user.

Reply:
`compilation_info` also not very ergonomic. I may not be stressing the point enough here. None of these flags (or even `compilation_info` for that matter) is for user experience. They are all meant to help with development only (`compilation_info` gives a bit more deployment-related support, but its literally a solution cause of lack of finding any other solution). Doing everything dynamically is the solution that needs to be done, but thats beside the point though. This is not related to user experience at all, and trying to fine tune it for user experience seems off.

Reply:
great, so we're in agreement about the destination.  let's discuss this PR's specific diff in a call tomorrow... it only introduces a couple of new options where there already were existing options, and unlike the existing options, the new options are much more intrinsically meaningful, in the sense that a user could set them to a specific value with a specific understanding of why that this is the right choice for their target workload details, while the existing option `defaultDistTileSize` was impossible to give an ideal value for all ops.

Reply:
It might be a semantic thing. The phrasing of "user-centeric" or "user-friendly" makes me think this is being seen as something more than it should be. Basically never should these flags be used at the Shark python API level or any other deployment model. It can only be a developer flag. Let's chat more on call tomorrow)

Reply:
Filling in from offline discussion. I think both or on same page here. This is a flag for developer use and using it anywhere outside of developer workflow is unsupported. So this isnt "supported" and shouldnt be used in branchmark tracking/deployment. A comment to that affect might help set expectations.